packages/babel-parser/test/helpers/to-fuzzed-options.js

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable no-confusing-arrow */
```

Comment:
Sure, is there a workaround other than "don't use arrow functions or don't use ternary operators"? I had turned it on since the eslint docs tell you that this warning wants you to wrap it in parenthesis to resolve the "ambiguity"... only to then complain about un-needed parenthesis.

Reply:
Sorry, to clarify, when I fix the arrow issue, then *prettier* complains about the disambiguating parenthesis. So it's a conflict between these two almost, but not quite, overlapping tools want.

Reply:
OK, so it looks like I'll just have to change like 20 arrows function to non-arrow functions, since what would be the result from autofixing this (`--fix`) is in conflict with the separate Prettier run. I'll do this if we feel strongly about it, but I'd perhaps like to make an appeal for instead turning this setting off (or being OK with having this comment disabling it here):

1. It's actually off by default in `eslint:recommended`, because as I'll describe below, it's kind of a stretch of an "ambiguity".
2. As described above, it's normally a simple fix (just wrapping in parenthesis), but in our case leads to bouncing back and forth between conflicting style enforcers.
3. The actual case being protected against being "confusing" is IMO somewhat... strange. The idea is to protect against the possibility that the author may have meant "<=" instead of "=>". So, for example, in my function below:

<img width="850" alt="Screen Shot 2022-01-26 at 6 59 47 PM" src="https://user-images.githubusercontent.com/23753/151268786-d6e6aada-d946-40d6-95e2-a10439e1a4da.png">

The idea is that perhaps I *actually* meant:

```javascript
const toExplanationString = ({ discrepancy, expected, actual, key }) <= discrepancy === "length"
```

In other words, the implication is that it's hard to tell whether I *meant* what I typed, an arrow function, or whether what I actually meant was to check whether a number-coerced newly created object (which would be `NaN`) is smaller than or equal to... the boolean result of a triple equals. Seems... unlikely. Especially as we move to typescript across the board.

Anyways, don't meant to be difficult, I'll go and convert them all to non-arrow functions (which unfortunately changes quite a few one-liners to 5-line noisier functions due the indented return then forces another wrap, etc.), but just wanted to throw it out there considering the weird conflict with Prettier, and on the off chance that this was not some super-important linting option but perhaps simply copied over form some other project or something.

Reply:
Good point! I believe `no-confusing-arrow` rule should allow such cases, at least behind an option. I have open https://github.com/eslint/eslint/issues/15548

Before ESLint team decide, I am good to leave it as-is.

Reply:
Great, thanks!